[kaidjohnson] Fix up dateparser support for 12 hour datetime format.

kaidjohnson opened the pull request:
Using a date format that includes a 12-hour time format (eg. dd MMMM yyyy hh:mma) can result in a permanent invalid parsing of the value.

I've created a plunk to demonstrate the issue (try manually deleting the 'M' from 'PM' then putting the 'M' back. It becomes invalid without the M but never is valid again when the M is restored.):
http://plnkr.co/edit/CvJgnoCE08XGlDbVQka8?p=preview

Here's the same plunkr, but with the fix found in this PR:
http://plnkr.co/edit/uTKEjRFcY3CsWDHyZ0bZ?p=preview

icfantv commented:
@kaidjohnson please add the relevant functional tests for this and update/rebase your PR.  thanks.

wesleycho commented on src/dateparser/dateparser.js:
```diff
@@ -53,12 +53,16 @@ angular.module('ui.bootstrap.dateparser', [])
     },
     'mm': {
-      regex: '[0-5][0-9]',
+      regex: '[0-5]{1}[0-9]{1}',
```
Revert this change - this is no different than the previous version, as tested [here](http://regexpal.com/).

wesleycho commented on src/dateparser/dateparser.js:
```diff
@@ -53,12 +53,16 @@ angular.module('ui.bootstrap.dateparser', [])
       apply: function(value) { this.hours = +value; }
     },
+    'hh': {
+      regex: '0[0-9]{1}|1[0-2]{1}',
```
Remove `{1}` in both instances in this line, they are unnecessary.

kaidjohnson commented:
Thanks @icfantv and @wesleycho -- I've added tests, removed redundant {1}, and corrected AM/PM parsing issue discovered during tests. Also updated plunkr

wesleycho commented:
LGTM - good work!
